### ISSUE TITLE
traefik stuck waiting on cos layer: waiting for service: 'traefik'

### ISSUE THREAD

**Issue by Basdbruijne** [canonical/operator#821]
In testrun https://solutions.qa.canonical.com/testruns/testRun/ba7598fc-8ad6-4a8d-a847-90f9ff321606, the cos layer doesn't come up on aws because traefik is stuck waiting:

```
Unit             Workload  Agent  Address          Ports  Message
alertmanager/0*  active    idle   192.168.88.137          
grafana/0*       active    idle   192.168.29.8            
loki/0*          active    idle   192.168.221.138         
prometheus/0*    active    idle   192.168.88.138          
traefik/0*       waiting   idle   192.168.221.137         waiting for service: 'traefik'
```

The crashdump for traefik is empty, so I can't provide more information at this point.

Link to all the crashdumps:
https://oil-jenkins.canonical.com/artifacts/ba7598fc-8ad6-4a8d-a847-90f9ff321606/index.html

**Comment by Basdbruijne** [comment]
There is some more information in https://oil-jenkins.canonical.com/artifacts/6fdb1794-c22d-4b8a-ae43-a5c1fb3140b8/generated/generated/cos/juju-crashdump-cos-2022-08-25-05.49.27.tar.gz in the debug log:

It looks like traefik got a weird address from AWS and can't handle it:
```
Traceback (most recent call last):
  File "./src/charm.py", line 678, in <module>
    main(TraefikIngressCharm)
  File "/var/lib/juju/agents/unit-traefik-0/charm/venv/ops/main.py", line 431, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-traefik-0/charm/venv/ops/main.py", line 142, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-traefik-0/charm/venv/ops/framework.py", line 316, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-traefik-0/charm/venv/ops/framework.py", line 784, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-traefik-0/charm/venv/ops/framework.py", line 857, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-traefik-0/charm/lib/charms/prometheus_k8s/v0/prometheus_scrape.py", line 1576, in _set_unit_ip
    unit_ip = str(self._charm.model.get_binding(relation).network.bind_address)
  File "/var/lib/juju/agents/unit-traefik-0/charm/venv/ops/model.py", line 679, in network
    self._network = self._network_get(self.name, self._relation_id)
  File "/var/lib/juju/agents/unit-traefik-0/charm/venv/ops/model.py", line 672, in _network_get
    return Network(self._backend.network_get(name, relation_id))
  File "/var/lib/juju/agents/unit-traefik-0/charm/venv/ops/model.py", line 724, in __init__
    self.ingress_addresses.append(ipaddress.ip_address(address))
  File "/usr/lib/python3.8/ipaddress.py", line 53, in ip_address
    raise ValueError('%r does not appear to be an IPv4 or IPv6 address' %
ValueError: 'a3df8cc229ead48859dc4c2726fd81bb-1924570860.us-east-1.elb.amazonaws.com' does not appear to be an IPv4 or IPv6 address
```

**Comment by PietroPasotti** [comment]
duplicate of https://github.com/canonical/operator/issues/821

**Comment by Basdbruijne** [comment]
Do you mean duplicate of #818 ?

**Comment by PietroPasotti** [comment]
yes, sorry :)
